### PR TITLE
Curate api trx append

### PR DESCRIPTION
**Pull request by msuhovec**

### CODE REVIEW

File: app/controllers/api/uploads_controller.rb [hunk 1]
```diff
@@ -37,39 +45,54 @@ def trx_new_file
       content = s3.bucket(ENV['S3_BUCKET']).object("#{trx_id}/#{file_pid}-#{next_sequence}")
       #copy body of message to bucket
       content.put(body: request.body())
-      metadata = s3.bucket(ENV['S3_BUCKET']).object("#{trx_id}/metadata-#{file_pid}.json")
-      metadata.put(body: initial_file_metadata(file_name, file_pid))
+      metadata = s3.bucket(ENV['S3_BUCKET']).object("#{trx_id}/metadata-file-#{file_pid}.json")
```
Comment (jeremyf) [inline app/controllers/api/uploads_controller.rb]: I see duplication in line 48 and line 17. Would it be feasible to extract a method to isolate some of that logic?

My personal foible says "Keep as much logic out of the controller" but I'm not prepared to die on that hill. I'm wondering if it makes sense to extract an object that handles much of this logic?

Comment (laritakr) [reply app/controllers/api/uploads_controller.rb]: We are trying to just get things working and then encapsulate the duplication. I'd prefer to leave it as is for now.

Comment (laritakr) [reply app/controllers/api/uploads_controller.rb]: But yes, long-term, the goal is to extract an object.

Comment (msuhovec) [reply app/controllers/api/uploads_controller.rb]: Larita and I have gone back and forth on  the controller vs app/model thing (well, maybe app/services in this case). I think we will end up doing that.

Comment (msuhovec) [reply app/controllers/api/uploads_controller.rb]: I have talked about consolidating the S3 stuff- I will do that now.

Comment (jeremyf) [reply app/controllers/api/uploads_controller.rb]: Excellent. Looks functional and I'm glad to hear that you're having conversations about approaches.

**Pull request merged**